projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -4,128 +4,186 @@
+      updateLWWMapForRequests(command.coordinates.asKey, command.requestedStatus)
+
+    // Bare Status come from worker and contain an observed Status
+    case observedStatus: Status =>
+      log.debug(s"Observed [${sender().path.toString}] as $observedStatus.")
+      reversedActorIndex.get(sender()) match {
+        case Some(workerName) => updateLWWMapForObserved(workerName, observedStatus)
+        case None             => log.error(s"Unknown actor [${sender().path.toString}] reports status $observedStatus.")
+      }
+
+    // TODO: handle UpdateSuccess/UpdateFailure !!
+    case UpdateSuccess(_, _) => //TODO
+    case _: UpdateFailure[_] => //TODO
+    case changed @ Changed(RequestedStatusDataKey) => {
+      val remotelyChanged                 = changed.get(RequestedStatusDataKey).entries
+      val diffs: Set[(WorkerKey, Status)] = remotelyChanged.toSet.diff(requestedStatusLocalCopy.toSet)
+
+      // when the requested status changes, we must forward the new value to the appropriate actor
+      // if it's a one of the workers in the local actorIndex
+      diffs
+        .foreach {
+          case (workerName, requestedStatus) =>
+            log.debug(s"Remotely requested worker [$workerName] as [$requestedStatus].")
+            actorIndex.get(workerName).foreach { workerRef =>
+              log.debug(
+                s"Setting requested status [$requestedStatus] on worker $workerName [${workerRef.path.toString}]"
+              )
+              workerRef ! requestedStatus
+            }
+        }
+
+      requestedStatusLocalCopy = remotelyChanged
+    }
+
+    // TODO: handle UpdateSuccess/UpdateFailure !!
+    case changed @ Changed(ObservedStatusDataKey) =>
+      observedStatusLocalCopy = changed.get(ObservedStatusDataKey).entries
 
-    case g @ GetSuccess(DataKey, req) =>
-      val registry: PNCounterMap[WorkerMetadata] = g.get(DataKey)
-      val desiredStatus: DesiredState            = mapStatus(registry.entries)
-      req.get.asInstanceOf[ActorRef] ! desiredStatus
+    // TODO: handle UpdateSuccess/UpdateFailure !!
+    case changed @ Changed(NameIndexDataKey) =>
+      nameIndexLocalCopy = changed.get(NameIndexDataKey).entries
 
     case Terminated(deadActor) =>
-      // update indices and stop watching
-      actorIndex = actorIndex - actorReverseIndex(deadActor)
-      actorReverseIndex = actorReverseIndex - deadActor
+      log.debug(s"Worker ${deadActor.path.name} died. Marking it as Stopped.")
+      // when a watched actor dies, we mark it as stopped. It will eventually
+      // respawn (thanks to EnsureActive) and come back to it's requested status.
+      reversedActorIndex.get(deadActor).foreach { name =>
+        updateLWWMapForObserved(name, Stopped)
+      }
+      // ... and then update indices and stop watching
+      actorIndex = actorIndex - reversedActorIndex(deadActor)
+      reversedActorIndex = reversedActorIndex - deadActor
       context.unwatch(deadActor)
```

Comment:
It doesn't make sense to unwatch a dead actor

Reply:
Hmmm, good point. I was being overcautious here and didn't want to assume `watched` actors that have terminate would not have an impact in performance. I don't know if and how much an actor with millions of watches may degrade so I thought `unwatch`ing wouldn't harm. Happy to remove the operation if it has no impact (happens already under the covers).

Reply:
We can check akka code for it, I would be surprised if there is such a leak. And if so, let's fix it there. 

I've never seen code that unwatch a dead actor, nor have a see any akka best practices saying so. So, I think we can be pretty much sure that this is not needed.

Reply:
No leak

https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala#L73-L89